tests/Integration/Behaviour/Features/Context/CurrencyFeatureContext.php

```diff
@@ -74,13 +74,15 @@ public function thereIsACurrency($currencyName, $currencyIsoCode, $changeRate)
         if (!$currencyId) {
             $currency = new Currency();
             $currency->name = $currencyIsoCode;
+            $currency->precision = 2;
```

Comment:
Any reason to set 2 here and having 4 in behat tests?

Reply:
Or even sometimes 3 :thinking:

Reply:
Probably because for now the CLDR uses the pattern to determine the number of decimals Not sure about it though because https://github.com/PrestaShop/PrestaShop/pull/15755 seems to use the currency precision Anyway this issue https://github.com/PrestaShop/PrestaShop/issues/14709 will deal with this problem more in details with appropriate Behat tests

But in the end I think the tests you are talking about manage the computing, so they are based on ComputingPrecision which by default is `3 * currency.precision => 6` That's why we have more decimals in the tests now The CLDR only applies on display so it's not the same value as for the 3 or 4 number of digits I guess it's only about the value

@matthieu-rolland correct me if I'm wrong on this one

Reply:
Better with explanations :D

Reply:
exactly @jolelievre 👍